homeassistant/components/sensor/wunderground.py

```diff
@@ -0,0 +1,164 @@
+"""Support for Wunderground weather service."""
+from datetime import timedelta
+import logging
+import requests
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.config_validation import ensure_list
+from homeassistant.util import Throttle
+from homeassistant.const import (CONF_PLATFORM, CONF_MONITORED_CONDITIONS,
+                                 CONF_API_KEY, TEMP_FAHRENHEIT, TEMP_CELSIUS,
+                                 STATE_UNKNOWN)
+
+CONF_PWS_ID = 'pws_id'
+
+_URL_QUERY = '/conditions/q/'
+_RESOURCE = 'http://api.wunderground.com/api/'
+_LOGGER = logging.getLogger(__name__)
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+
+# Sensor types are defined like: Name, units
+SENSOR_TYPES = {
+    'weather': ['Weather Summary', None],
+    'station_id': ['Station ID', None],
+    'feelslike_c': ['Feels Like (°C)', TEMP_CELSIUS],
+    'feelslike_f': ['Feels Like (°F)', TEMP_FAHRENHEIT],
+    'feelslike_string': ['Feels Like', None],
+    'heat_index_c': ['Dewpoint (°C)', TEMP_CELSIUS],
+    'heat_index_f': ['Dewpoint (°F)', TEMP_FAHRENHEIT],
+    'heat_index_string': ['Heat Index Summary', None],
+    'dewpoint_c': ['Dewpoint (°C)', TEMP_CELSIUS],
+    'dewpoint_f': ['Dewpoint (°F)', TEMP_FAHRENHEIT],
+    'dewpoint_string': ['Dewpoint Summary', None],
+    'wind_kph': ['Wind Speed', 'kpH'],
+    'wind_mph': ['Wind Speed', 'mpH'],
+    'UV': ['UV', None],
+    'pressure_in': ['Pressure', 'in'],
+    'pressure_mb': ['Pressure', 'mbar'],
+    'wind_dir': ['Wind Direction', None],
+    'wind_string': ['Wind Summary', None],
+    'temp_c': ['Temperature (°C)', TEMP_CELSIUS],
+    'temp_f': ['Temperature (°F)', TEMP_FAHRENHEIT],
+    'relative_humidity': ['Relative Humidity', '%'],
+    'visibility_mi': ['Visibility (miles)', 'mi'],
+    'visibility_km': ['Visibility (km)', 'km'],
+    'precip_today_in': ['Precipation Today', 'in'],
+    'precip_today_metric': ['Precipitation Today', 'mm'],
+    'precip_today_string': ['Precipitation today', None],
+    'solarradiation': ['Solar Radiation', None]
+}
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): "wunderground",
+    vol.Required(CONF_API_KEY): vol.Coerce(str),
+    CONF_PWS_ID: vol.Coerce(str),
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]): ensure_list,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Wunderground sensor."""
+    if not PLATFORM_SCHEMA(config):
+        return False
+
+    rest = WUndergroundData(hass,
+                            config.get(CONF_API_KEY),
+                            config.get(CONF_PWS_ID, None))
+    sensors = []
+    for variable in config['monitored_conditions']:
+        if variable in SENSOR_TYPES:
```

Comment:
Made a PR to address concerns Fabian had.

On Aug 18, 2016 10:28 AM, "Paulus Schoutsen" <notifications@github.com>
wrote:

> In homeassistant/components/sensor/wunderground.py
> <https://github.com/home-assistant/home-assistant/pull/2861#discussion_r75341586>
> :
>
> > +    CONF_PWS_ID: vol.Coerce(str),
> > +    vol.Required(CONF_MONITORED_CONDITIONS, default=[]): ensure_list,
> > +})
> > +
> > +
> > +def setup_platform(hass, config, add_devices, discovery_info=None):
> > +    """Setup the Wunderground sensor."""
> > +    if not PLATFORM_SCHEMA(config):
> > +        return False
> > +
> > +    rest = WUndergroundData(hass,
> > +                            config.get(CONF_API_KEY),
> > +                            config.get(CONF_PWS_ID, None))
> > +    sensors = []
> > +    for variable in config['monitored_conditions']:
> > +        if variable in SENSOR_TYPES:
>
> I don't think this remove was committed
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/2861/files/a5fd04f2155ca7505143359293a5d7718f6b09e9#r75341586>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZS0pvY-ORNtCdNDLw4pNYj0rtEMtks5qhIhAgaJpZM4Jm1ur>
> .
>